go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -450,28 +450,119 @@ func handleNetdevResources(resourceName string) (string, error) {
+		return "", err
+	}
+	klog.Infof("Imported pfId '%v' and FuncId '%v' for node '%v'", pfId, vfId, node.Name)
+
+	return util.GetSriovnetOps().GetVfRepresentorDPU(fmt.Sprintf("%d", pfId), fmt.Sprintf("%d", vfId))
+}
+
+// Take care of alternative names for the netdevName by making sure we
+// use the link attribute name as well as handle the case when netdevName
+// was renamed to types.K8sMgmtIntfName
+func getManagementPortNetDev(netdevName string) (string, error) {
+	link, err := util.GetNetLinkOps().LinkByName(netdevName)
+	if err != nil {
+		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
+			return "", fmt.Errorf("failed to lookup %s link: %v", netdevName, err)
+		}
+		// this may not the first time invoked on the node after reboot
+		// netdev may have already been renamed to ovn-k8s-mp0.
+		link, err = util.GetNetLinkOps().LinkByName(types.K8sMgmtIntfName)
 		if err != nil {
-			// Device might had been already renamed to types.K8sMgmtIntfName
-			config.OvnKubeNode.MgmtPortNetdev = types.K8sMgmtIntfName
-			if deviceID, err = util.GetDeviceIDFromNetdevice(config.OvnKubeNode.MgmtPortNetdev); err != nil {
-				return nil, nil, fmt.Errorf("failed to get device id for %s or %s: %v",
-					config.OvnKubeNode.MgmtPortNetdev, types.K8sMgmtIntfName, err)
+			return "", fmt.Errorf("failed to get link device for %s. %v", netdevName, err)
+		}
+	}
+
+	if link.Attrs().Name != netdevName {
+		klog.Infof("'%v' != '%v' (link.Attrs().Name != netdevName)", link.Attrs().Name, netdevName)
+	}
+	return link.Attrs().Name, err
+}
+
+// If netdevice name is not provided in the full mode then management port backed
+// by OVS internal port. If it is provided then it is backed by VF or SF and need
+// to determine its representor name to plug into OVS integrational bridge
+// In case ovn-k is running in DPU mode, it will read the annotation from the host
+// side which should have been exported by ovn-k running in DPU host mode.
+func getMgmtPortAndRepName(node *kapi.Node) (string, string, error) {
+	switch config.OvnKubeNode.Mode {
+	case types.NodeModeFull:
+		if config.OvnKubeNode.MgmtPortNetdev != "" {
+			var err error
+			netdevName := config.OvnKubeNode.MgmtPortNetdev
+			netdevName, err = getManagementPortNetDev(netdevName)
+			if err != nil {
+				return "", "", err
 			}
+			deviceID, err := util.GetDeviceIDFromNetdevice(netdevName)
+			if err != nil {
+				return "", "", fmt.Errorf("failed to get device id for %s: %v", netdevName, err)
+			}
+			rep, err := util.GetFunctionRepresentorName(deviceID)
+			if err != nil {
+				return "", "", err
+			}
+			return netdevName, rep, err
+		} else {
+			return config.OvnKubeNode.MgmtPortNetdev, "", nil
+		}
+	case types.NodeModeDPU:
+		rep, err := importManagementPortAnnotation(node)
+		if err != nil {
+			return "", "", err
+		}
+		return "", rep, err
+	case types.NodeModeDPUHost:
+		var err error
+		netdevName := config.OvnKubeNode.MgmtPortNetdev
+		netdevName, err = getManagementPortNetDev(netdevName)
```

Comment:
Pass `config.OvnKubeNode.MgmtPortNetdev` directly instead of assigning it to the variable.